cmd/zfs/zfs_main.c

```diff
@@ -6731,7 +6732,7 @@ zfs_do_holds(int argc, char **argv)
 typedef struct get_all_state {
 	boolean_t	ga_verbose;
 	get_all_cb_t	*ga_cbp;
-	const char	*ga_dataset;
+	const char	*ga_filesystem;
```

Comment:
This should really be called `ga_parent` or something similar, to make it clear that the state is a desired *parent* rather than a desired *filesystem*. I only recommend using the `filesystem` variable name in the `share_mount` function, where you are using it to refer to the "filesystem" argument.

Reply:
Good point, `parent` is more descriptive/accurate.